helm/charts/kratos/templates/statefulset-mail.yaml

```diff
@@ -4,6 +4,7 @@
 {{- $labels := ternary .Values.statefulSet.labels .Values.deployment.labels (not (empty .Values.statefulSet.labels )) -}}
 {{- $extraVolumeMounts := ternary .Values.statefulSet.extraVolumeMounts .Values.deployment.extraVolumeMounts (not (empty .Values.statefulSet.extraVolumeMounts )) -}}
 {{- $extraEnv := ternary .Values.statefulSet.extraEnv .Values.deployment.extraEnv (not (empty .Values.statefulSet.extraEnv )) -}}
+{{- $environmentSecretsName := ternary .Values.statefulSet.environmentSecretsName .Values.deployment.environmentSecretsName (not (empty .Values.statefulSet.environmentSecretsName )) -}}
```

Comment:
IF this value is supposed to be defined only for the courier, then we don't need this part and can use it like `.Values.statefulSet.environmentSecretsName`

Reply:
It is not dedicated to be only defined for statefulSet. The main goal is reusing environments of the the deployment. But above code allow to define different environment for  statefulSet.